lldb/test/API/functionalities/data-formatter/type-recognizer/TestTypeRecognizerList.py

```diff
@@ -0,0 +1,21 @@
+"""
+Test lldb data formatter subsystem.
+"""
+
+
+from lldbsuite.test.lldbtest import *
+
+
+class TypeRecognizerListTestCase(TestBase):
+    def test_type_recognizer_list(self):
+        """Test that the 'type recognizer list' command handles command line arguments properly"""
+        self.expect(
+            "type recognizer list Foo", substrs=["Category: default", "Category: system"]
+        )
+
+        self.expect("type recognizer list -w default", substrs=["system"], matching=False)
```

Comment:
Both of the tests you have here use `matching=False`. It's not clear what output you expect here